internal/acceptance/helpers/check_destroyed.go

```diff
@@ -10,25 +10,25 @@ import (
 )
 
 // CheckDestroyedFunc returns a TestCheckFunc which validates the resource no longer exists
-func CheckDestroyedFunc(client *clients.Client, testResource types.TestResource, resourceType, resourceLabel string) func(state *terraform.State) error {
+func CheckDestroyedFunc(client *clients.Client, testResource types.TestResource, resourceType, resourceName string) func(state *terraform.State) error {
 	return func(state *terraform.State) error {
 		ctx := client.StopContext
 
 		for label, resourceState := range state.RootModule().Resources {
 			if resourceState.Type != resourceType {
 				continue
 			}
-			if label != resourceLabel {
+			if label != resourceName {
 				continue
 			}
 
 			// Destroy is unconcerned with an error checking the status, since this is going to be "not found"
 			result, err := testResource.Exists(ctx, client, resourceState.Primary)
-			if result == nil && err != nil {
-				return fmt.Errorf("should have either an error or a result when checking if \"%s.%s\" has been destroyed", resourceType, resourceLabel)
+			if result == nil && err == nil {
+				return fmt.Errorf("should have either an error or a result when checking if %q has been destroyed", resourceName)
 			}
 			if result != nil && *result {
-				return fmt.Errorf("\"%s.%s\" still exists", resourceType, resourceLabel)
+				return fmt.Errorf("%q still exists", resourceName)
```

Comment:
TestData.ResourceTest() has been passing in the full resource name (`foo.bar`) for the 4th argument rather than just the label (`bar`), which is why I've gone for renaming the argument - previously this would have outputted something like "foo.foo.bar still exists".